### ISSUE TITLE
cppdbg `miDebuggerServerAddress` parameter generates warning

### ISSUE THREAD

**Issue by zfields** [microsoft/vscode-cpptools#9195]
Using `  "miDebuggerServerAddress": "localhost:3333"` seems to have the same effect as setting `target remote :3333` in the `postRemoteConnectCommands`, which generates the following warning:

```none
openocd: Warn : Prefer GDB command "target extended-remote 3333" instead of "target remote 3333"
```

However, setting `target extended-remote :3333` in the `postRemoteConnectCommands` and removing `miDebuggerServerAddress` eliminates the warning.

Is this an oversight, or is it by design?

I'm using an ESP32 WROOM-32D and the ESP Prog debugger, as well as the `openocd` and `gdb` provided by the latest version of ESP-IDF.

*'launch.json'*
```json
{
	"name": "ESP-Prog Debug (C/C++)",
	"type": "cppdbg",
	"request": "launch",
	"program": "${workspaceRoot}/Radar/build/esp_prog_vscode_debug.elf",
	"args": [],
	"cwd": "${workspaceRoot}/Radar",
	"environment": [
		{
			"name": "PATH",
			"value": "${config:esp_toolchain_paths}:${env:PATH}"
		}
	],
	"externalConsole": false,
	"MIMode": "gdb",
	"miDebuggerPath": "${config:esp_gdb_path}/xtensa-esp32-elf-gdb",
	"debugServerPath": "${config:esp_openocd_path}/openocd",
	"debugServerArgs": "-f interface/ftdi/esp32_devkitj_v1.cfg -f target/esp32.cfg -c \"esp32.cpu1 configure -rtos FreeRTOS\"",
	"serverStarted": "Listening on port [0-9]+ for gdb connections",
	"filterStderr": true,
	"filterStdout": false,
	"launchCompleteCommand": "None",
	"postRemoteConnectCommands": [
		{
			"description": "Target Remote Device on Port 3333",
			"text": "target extended-remote :3333",
			"ignoreFailures": false
		},
		{
			"description": "Respect Hardware Limitations (as prescribed by Espressif)",
			"text": "set remote hardware-watchpoint-limit 2",
			"ignoreFailures": false
		},
		{
			"description": "Hard Reset and Immediately Halt",
			"text": "monitor reset halt",
			"ignoreFailures": false
		},
		{
			"description": "Flush Internal Register Cache",
			"text": "flushregs",
			"ignoreFailures": false
		},
		{
			"description": "Set Temporary Hardware Assisted Breakpoint at `call_start_cpu0`",
			"text": "thbreak call_start_cpu0",
			"ignoreFailures": false
		},
		{
			"description": "Set Temporary Hardware Assisted Breakpoint at `app_main`",
			"text": "thbreak app_main",
			"ignoreFailures": false
		},
		{
			"description": "Shutdown GDB Server on GDB Detach",
			"text": "monitor [target current] configure -event gdb-detach { shutdown }",
			"ignoreFailures": false
		},
	],
	"stopAtConnect": false,
	"logging": {
		"exceptions": true,
		"engineLogging": false,
		"moduleLoad": true,
		"programOutput": true,
		"trace": true,
		"traceResponse": false
	},
},
```

**Comment by WardenGnaw** [comment]
There should be an option for `"useExtendedRemote": true` that we added in https://github.com/microsoft/MIEngine/pull/1260.

@xisui-MSFT Since you added this flag, can you see if this is an issue?

**Comment by fbs2016** [comment]
Any update? I meet similar issue when try extended remote debug. The option "useExtendedRemote" has been delivered in latest cpptools V1.10.7? I can't add this option in launch.json?

**Comment by sean-mcmanus** [comment]
The useExtendedRemote option is available in 1.11.3 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.11.3
